bot.py

```diff
@@ -41,122 +40,207 @@
+
+    @property
+    def main_category(self):
+        if self.guild:
+            return discord.utils.get(self.modmail_guild.categories, name='Mod Mail')
+
+    @property
+    def blocked_users(self):
+        if self.modmail_guild:
+            top_chan = self.main_category.channels[0]
+            return [int(i) for i in re.findall(r'\d+', top_chan.topic)]
+    
+    @property
+    def prefix(self):
+        return self.config.get('prefix', '?')
+
     @staticmethod
     async def get_pre(bot, message):
         '''Returns the prefix.'''
-        p = bot.config.get('PREFIX') or 'm.'
-        return [p, f'<@{bot.user.id}> ', f'<@!{bot.user.id}> ']
-    
-    def owner_only():
-        async def predicate(ctx):
-            allowed = [int(x) for x in ctx.bot.config.get('OWNERS', '0').split(',')]
-            return ctx.author.id in allowed
-        return commands.check(predicate)
-    
-    def trigger_typing(func):
-        @functools.wraps(func)
-        async def wrapper(self, ctx, *args, **kwargs):
-            await ctx.trigger_typing()
-            return await func(self, ctx, *args, **kwargs)
-        return wrapper
+        return [bot.prefix, f'<@{bot.user.id}> ', f'<@!{bot.user.id}> ']
 
     async def on_connect(self):
         print(line)
-        print(Fore.YELLOW + 'Connected to gateway.')
-        
-        self.session = aiohttp.ClientSession()
-        status = os.getenv('STATUS') or self.config.get('STATUS')
+        print(Fore.CYAN + 'Connected to gateway.')
+
+        status = self.config.get('status')
         if status:
             await self.change_presence(activity=discord.Game(status))
 
-    @property
-    def guild_id(self):
-        return int(self.config.get('GUILD_ID'))
-    
-    @property
-    def guild(self):
-        g = discord.utils.get(self.guilds, id=self.guild_id)
-        return g
-
     async def on_ready(self):
         '''Bot startup, sets uptime.'''
         print(textwrap.dedent(f'''
         {line}
-        {Fore.YELLOW}Client ready.
+        {Fore.CYAN}Client ready.
         {line}
-        {Fore.YELLOW}Logged in as: {self.user}
-        {Fore.YELLOW}User ID: {self.user.id}
-        {Fore.YELLOW}Guild ID: {self.guild.id if self.guild else 0}
+        {Fore.CYAN}Logged in as: {self.user}
+        {Fore.CYAN}User ID: {self.user.id}
+        {Fore.CYAN}Guild ID: {self.guild.id if self.guild else 0}
         {line}
         ''').strip())
+        
+        await self.threads.populate_cache()
+
+    async def process_modmail(self, message):
+        '''Processes messages sent to the bot.'''
+
+        reaction = '🚫' if message.author.id in self.blocked_users else '✅'
+
+        try:
+            await message.add_reaction(reaction)
+        except:
+            pass
+
+        blocked_em = discord.Embed(
+            title='Message not sent!', 
+            color=discord.Color.red(),
+            description='You have been blocked from using modmail.'
```

Comment:
Are we sure we want to inform end users if they are blocked? Or just a silent ignore?

Reply:
Informing them is fine, silent ignore may make them think the bot is broken or something.